R/compute_duration.R

```diff
@@ -154,16 +154,23 @@ compute_duration <- function(start_date,
   # Checks
   assert_date_vector(start_date)
   assert_date_vector(end_date)
-  assert_character_scalar(in_unit, values = valid_time_units())
+  assert_character_scalar(in_unit, values = c(
+    valid_time_units(),
+    toupper(valid_time_units())
+  ))
   assert_character_scalar(type, values = c("interval", "duration"))
   assert_character_scalar(out_unit, values = c(
-    valid_time_units(), "weeks",
-    "min", "sec"
+    valid_time_units(),
+    toupper(valid_time_units()),
+    "weeks", "min", "sec"
```

Comment:
What about values like `"year"`, `"yrs"`, `"wk"`, `"month"`, ...?

Maybe option 1 was not a good decision. This could become a source of frequent updates.

Reply:
I thought we were only updating to the ones allowed in the CDSIC image?

Reply:
Yeah maybe we stick to those just to cover our bases with regards to this codelist

Reply:
I think it is not mandatory to use the codelist and the codelist is extensible. So users may want to use different values for the units and ask us to update the function. I wonder if we should update the function such that any value can be used.

Reply:
Perhaps we have a soft check/message on the values that are shown in the codelist with an option to allow an extensible codelist.  

So a user would be prompted with a message that says "These are the common values used in this function.  If you wish to use additional values please mark `extensible = TRUE`  to suppress this message

Reply:
I think this will not work because we use the `out_unit` value not only for setting the unit variable but also for calling lubridate functions. So we need to translate these values before passing them to the lubridate function.

Reply:
I think where it gets a little tricky is we rely on passing whatever we deem as acceptable unit values into the `unit` argument inside the `lubridate` package, which after investigating a bit, they have some pretty "loose" acceptable terms (they generally accept the singular or plural versions of the whole word of the first three letters, in particular see second screenshot for clear demo)

![image](https://github.com/pharmaverse/admiral/assets/43018762/73ac68b5-7ad8-4715-84d2-c75379fa56e1)

![image](https://github.com/pharmaverse/admiral/assets/43018762/9122e42a-4d39-4bdc-9dd0-f661fe8bf651)

My proposal if we want to assist in mapping to the proper lubridate functions we limit our scope to:

- the standard ones (case-agnostic) singular & plural forms so week, WEEK, etc. 
- Y, W, D, H, S (can't do M because month vs minutes) 
- yr(s), mo(s), wk(s), hr(s)